### PR TITLE
 lib.mkDirectory: replace python3.7 to sitePackages

### PR DESCRIPTION
**Pull request by GTrunSec**
This PR related my comment at https://github.com/tweag/jupyterWith/issues/124#issuecomment-694153591

- test with python38(recent nixpkgs)
```
wrapping `/nix/store/s1j7fk9sm91q4jzdv1747yrfqla31v51-python3.8-jupyterlab-2.2.4/bin/jupyter-lab'...
Rewriting #!/nix/store/685kq8pyhrvajah1hdsfn4q7gm3j4yd4-python3-3.8.5/bin/python3.8 to #!/nix/store/685kq8pyhrvajah1hdsfn4q7gm3j4yd4-python3-3.8.5
wrapping `/nix/store/s1j7fk9sm91q4jzdv1747yrfqla31v51-python3.8-jupyterlab-2.2.4/bin/jupyter-labextension'...
Rewriting #!/nix/store/685kq8pyhrvajah1hdsfn4q7gm3j4yd4-python3-3.8.5/bin/python3.8 to #!/nix/store/685kq8pyhrvajah1hdsfn4q7gm3j4yd4-python3-3.8.5
wrapping `/nix/store/s1j7fk9sm91q4jzdv1747yrfqla31v51-python3.8-jupyterlab-2.2.4/bin/jupyter-labhub'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
pythonCatchConflictsPhase
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
```
- test with python37 (current nixpkgs)
```
on3-3.7.7
wrapping `/nix/store/wfxfbym3ykw48a460gzcpgn1d5yx3h18-python3.7-jupyterlab-2.1.2/bin/jlpm'...
Rewriting #!/nix/store/nnfpapxzrhib9gxpj26f0i5h0ww9rfwc-python3-3.7.7/bin/python3.7 to #!/nix/store/nnfpapxzrhib9gxpj26f0i5h0ww9rfwc-python3-3.7.7
wrapping `/nix/store/wfxfbym3ykw48a460gzcpgn1d5yx3h18-python3.7-jupyterlab-2.1.2/bin/jupyter-lab'...
Rewriting #!/nix/store/nnfpapxzrhib9gxpj26f0i5h0ww9rfwc-python3-3.7.7/bin/python3.7 to #!/nix/store/nnfpapxzrhib9gxpj26f0i5h0ww9rfwc-python3-3.7.7
wrapping `/nix/store/wfxfbym3ykw48a460gzcpgn1d5yx3h18-python3.7-jupyterlab-2.1.2/bin/jupyter-labextension'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
pythonCatchConflictsPhase
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
```

### CODE REVIEW

**Comment by GTrunSec** [comment]
@guaraqe this PR for next time to update global nixpkgs.

File: lib/directory.nix [hunk 1]
```diff
@@ -18,7 +18,7 @@ in
         # consider that it comes from a folder without read access only as in
         # Nix
         mkdir -p "$DIRECTORY"/staging
-        cp ${jupyter}/lib/python3.7/site-packages/jupyterlab/staging/yarn.lock "$DIRECTORY"/staging
+        cp ${jupyter}/lib/${pkgs.python3.libPrefix}/site-packages/jupyterlab/staging/yarn.lock "$DIRECTORY"/staging
```
Comment (stolyaroleh) [inline lib/directory.nix:21]: Consider using `${pkgs.python3.sitePackages}/jupyterlab` instead of `${pkgs.python3.libPrefix}/site-packages/jupyterlab`.

Comment (GTrunSec) [reply lib/directory.nix]: Thanks

**Comment by GTrunSec** [comment]
nix-shell test 
```
terlab-2.2.8...                                                                                                                          
Rewriting #!/nix/store/cqh3nsk4z8lc960g8yqvz5nf2f6qx9v0-python3-3.8.6/bin/python3.8 to #!/nix/store/cqh3nsk4z8lc960g8yqvz5nf2f6qx9v0-pyth
on3-3.8.6                                                                                                                                
wrapping `/nix/store/qil7wms0pzixl2wxwwdnh0h9550nz8pn-python3.8-jupyterlab-2.2.8/bin/jlpm'...                                            
Rewriting #!/nix/store/cqh3nsk4z8lc960g8yqvz5nf2f6qx9v0-python3-3.8.6/bin/python3.8 to #!/nix/store/cqh3nsk4z8lc960g8yqvz5nf2f6qx9v0-pyth
on3-3.8.6                                                                                                                                
wrapping `/nix/store/qil7wms0pzixl2wxwwdnh0h9550nz8pn-python3.8-jupyterlab-2.2.8/bin/jupyter-lab'...                                     
Rewriting #!/nix/store/cqh3nsk4z8lc960g8yqvz5nf2f6qx9v0-python3-3.8.6/bin/python3.8 to #!/nix/store/cqh3nsk4z8lc960g8yqvz5nf2f6qx9v0-pyth
on3-3.8.6                                                                                                                                
wrapping `/nix/store/qil7wms0pzixl2wxwwdnh0h9550nz8pn-python3.8-jupyterlab-2.2.8/bin/jupyter-labextension'...                            
Rewriting #!/nix/store/cqh3nsk4z8lc960g8yqvz5nf2f6qx9v0-python3-3.8.6/bin/python3.8 to #!/nix/store/cqh3nsk4z8lc960g8yqvz5nf2f6qx9v0-pyth
on3-3.8.6                                                                                                                                
wrapping `/nix/store/qil7wms0pzixl2wxwwdnh0h9550nz8pn-python3.8-jupyterlab-2.2.8/bin/jupyter-labhub'...                                  
Executing pythonRemoveTestsDir                                                                                                           
Finished executing pythonRemoveTestsDir                                                                                                  
pythonCatchConflictsPhase                                                                                                                
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
```

**Comment by GTrunSec** [comment]
@guaraqe Please review this PR in priority. (Python3.8 already default in nixpkgs-21.05)

**Review by guaraqe — APPROVED** [review]
Sorry for the delay, thanks for this!

**Pull request merged**